zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/ExporterResourceProvider.java

```diff
@@ -16,7 +17,7 @@
 
 public interface ExporterResourceProvider {
 
-  void init(ExporterConfiguration configuration);
+  void init(ExporterConfiguration configuration, final ClientAdapter clientAdapter);
```

Comment:
❓ We actually just pass in here the client Adapter because at the end we want a cache loader. We had to add the cache loader to the ClientAdapter even. 

Couldn't we instead just pass in here the CacheLoader? And when calling the method we give a representive instantiation?

Reply:
It was not possible because the cache loader needs the process index name, which is set only in the provider.

Reply:
So what you actually need is a factory for the CacheLoader.

Reply:
Refactored this to use a factory.